File: michael/day_1/day_1.py
#day_1 AOC 2022 python file
import time
FULL_DATA, TEST_DATA = "./michael/day_1/day_1.txt", "./michael/day_1/day_1_min.txt"

def execute_time(func):
	def wrapper(*args, **kwargs):
		start = time.process_time()
		result = func(*args, **kwargs)
		end = time.process_time()
		print(f"{func.__name__}: {result}\nRan in {end - start} seconds")
		return result
	return wrapper

@execute_time
def part_one(calories):
	return max(calories)

@execute_time
def part_two(calories):
	calories.sort()
	return sum(calories[-3:])


with open(FULL_DATA) as f:
	calorie_counts = [sum([int(calorie) for calorie in calorie_group.split("\n")]) for calorie_group in f.read().split("\n\n")]

part_one(calorie_counts)
part_two(calorie_counts)

File: michael/day_1/day_1.js
//day_1 AOC 2022 with Node
const fs = require('fs');

const [fullData, testData] = ["./michael/day_1/day_1.txt", "./michael/day_1/day_1_min.txt"]

const executionTime = (func, data) =>{
    console.time(`${func.name} time:`)
    const result = func(data);
    console.timeEnd(`${func.name} time:`)
    console.log(`${func.name} Result: ${result}`)
    return result
}

const partOne = (calorieGroups) => {
    const arrSums = calorieGroups.map(x => x.reduce((a, b) => a + b))
    const max = arrSums.reduce((a, b) => Math.max(a, b), -Infinity);
    return max
}

const partTwo = (calorieGroups) => {
    const arrSums = calorieGroups.map(x => x.reduce((acc, current) => acc + current))
    const maxSorted = arrSums.sort((a,b) => a-b).slice(-3)
                           .reduce((a, b) => a + b) 
    return maxSorted
}

const buffer = fs.readFileSync(fullData);
let data = buffer.toString().split("\r\n\r");
data = data.map(x => x.split("\r\n").map(y => +y.replace("\n", "")))

executionTime(partOne, data)
executionTime(partTwo, data)


